src/pages/proposal.tsx

```diff
@@ -146,13 +143,20 @@ export const Proposal: React.FC = () => {
     data: proposal,
     error: proposalError,
     isLoading: proposalIsLoading,
-  } = useDaoProposal(
-    daoDetails?.address as string,
-    proposalId!,
-    pluginType,
-    pluginAddress,
-    intervalInMills
+    refetch,
+  } = useProposal(
+    {
+      pluginType: pluginType,
+      id: proposalId?.toString() ?? '',
+    },
+
+    {
+      refetchInterval: shouldRefetchOnInterval
```

Comment:
Can't we just check the `proposalStatus` here instead of using the `shouldRefetchOnInterval` state?
```suggestion
      refetchInterval: proposalStatus === ProposalStatus.ACTIVE
```

Reply:
That was my first thought, but turns out, I can't use `proposalStatus` because it hasn't been defined yet. Same with `proposal`, which makes sense since it's kind of a chicken and egg thing with `proposal` and `proposalStatus` being obtained by the very hook that requires it in its setup.

Reply:
You're right 🙈 Buut I just saw that react-query also accept a function for the `refetchInterval` parameter that takes as input the latest fetched data, can we try doing something like:
```
refetchInterval: (data) => data?.status === ProposalStatus.ACTIVE
```

Reply:
Sweet!